dex/tests/close-markets/programs/close-markets/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+use anchor_lang::prelude::*;
```

Comment:
Some more test scenarios that should be exercised: 

* closing a market immediately after opening
* closing a market with outstanding orders
* closing a market after trading and getting filled
* closing a market after trading and canceling
* closing a market with uncranked events
* closing a market with no outstanding orders, no uncranked events, but an open orders account that hasn't been settled. Should allow closing to reclaim rent exemption and subsequently settling